### ISSUE TITLE
Altmetrics badge

### ISSUE THREAD

**Issue by mdozmorov** [badges/shields#6110]
:clipboard: **Description**

[Altmetrics](https://www.altmetric.com/) - impact tracking of publications, preprints, etc.

Altmetrics badges show the impact of an associated DOI. 

:link: **Data**

[Examples and API for embedding](https://api.altmetric.com/embeds.html)

:microphone: **Motivation**

I maintain lists of scientific papers, such as [HiC tools](https://github.com/mdozmorov/HiC_tools). Knowing which paper is more impactful (from Altmetrics badges) will be immensely helpful to prioritize one's time.

**Comment by chris48s** [comment]
Just having a quick look at this, their badges depend on client-side logic, but there is a public API at https://api.altmetric.com/
Example call: https://api.altmetric.com/v1/doi/10.1038/480426a

By the looks of it, we would need to use an API key to avoid being rate limited. I'm slightly unclear where this project would sit as an API user because we definitely wouldn't be using the API for commercial use, but we're certainly not doing scientometrics research either..

**Comment by mdozmorov** [comment]
I'm not sure if it is possible to make a badge that can handle either non-registered API calls, or, if provided a user-specific key, be more flexible. This would make it user-specific.

Specifically, a badge with just Altmetrics counts would help. I use it non-registered, like [here](https://dozmorovlab.github.io/publications.html). Would it be possible to make such, even it will be rate limiting?

**Comment by calebcartwright** [comment]
The docs do mention:

> If you’d like to use the Altmetric Details Page API for non-commercial purposes that don’t exactly meet the “research project” use case, please email us at info@altmetric.com to discuss your options.

So I suppose we could send them a message from the Shields email. 

> Would it be possible to make such, even it will be rate limiting?

Possible, technically yes, but not something we'd just do before figuring out our strategy.

While I don't get the sense this would be one of our most popular badges, we still have to be cognizant of the fact that our badge servers send a _lot_ of traffic at the upstream platforms in many, many cases. These folks are pretty explicit in their docs:

> Please don't hammer the server! Where possible limit calls to once a second.

We've an obligation to be courteous consumers of the upstream service we integrate with, and it seems like we'll need to figure out how we want to handle the integration with this service; it's not something we'd want to skip and defer for later.

**Comment by mdozmorov** [comment]
Contacting Altmetrics from the Shields e-mail seems like the best strategy. 

Regarding popularity, I envision anyone with a science-related GitHub repository will want to have an Altmetrics badge for their papers listed in README.md. It is a matter of announcing on academic Twitter, and I'm sure the news about Altmetrics badges will spread momentarily and will be in high demand.

**Comment by tapanchudasama** [comment]
If anybody is not working on this and the things that are discussed above are solved, can I pick this up? @chris48s

**Comment by mdozmorov** [comment]
It will be great if the work on this can resume. Many tools citing their papers will benefit from Altmetrics badges, as well as collections of awesome papers and the like.

**Comment by chris48s** [comment]
I don't think we ever followed up with altmetrics because we didn't have anyone actively looking to work on the implementation but @tapanchudasama - if you're up for submitting a PR for this and have a clear idea what metrics you want to work on, I can chuck them an email.

**Comment by tapanchudasama** [comment]
@chris48s Yeah, I saw the sample response of the API. I think we can use the metric `at` that denotes Current Altmetric Attention Score in the badge. If this sounds good, then I can start working on this.
```
"history": {
  "1y": 0,
  "6m": 0,
  "3m": 0,
  "1m": 0,
  "1w": 0,
  "6d": 0,
  "5d": 0,
  "4d": 0,
  "3d": 0,
  "2d": 0,
  "1d": 0,
  "at": 292.904
},
```

**Comment by mdozmorov** [comment]
Suggesting the following look:
![image](https://user-images.githubusercontent.com/864945/113512678-b788d200-9533-11eb-8e98-08191dffe378.png)

In HTML, it can be inserted using:
````
<span data-badge-type="4" data-doi="10.3389/fbioe.2018.00198" data-hide-no-mentions="true" class="altmetric-embed"> </span> 
````

But HTML tags don't work in `README.md` files on GitHub. That's where Altmetrics shields will help.

And yes, the `"at"` field is the right one. The [API call for 10.3389/fbioe.2018.00198](https://api.altmetric.com/v1/doi/10.3389/fbioe.2018.00198) returns Altmetrics score in this field.

**Comment by chris48s** [comment]
I sent them an email. We'll see where it goes..

**Comment by tapanchudasama** [comment]
Any updates? @chris48s

**Comment by chris48s** [comment]
Nope. I got an automated confirmation from them. Then they made me sign up for their support portal and said someone would get back to me. Now their support thing seems to be broken (see if you can get as far as typing in a username on this login form before it force-redirects you: https://help.altmetric.com/support/login ). I'll check back and see if it gets fixed at some point :shrug:

**Comment by chris48s** [comment]
Nothing conclusive but I had an update today:

> Thank you for your email and apologies for the delay of answer. Our non-commercial API access is only offered to researchers within universities usually, but I've forwarded your request to the technical team who will be able to come back to you with more information.

Will keep the thread posted..
